client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx

```diff
@@ -15,7 +15,8 @@ const ChecklistItem = ( { task }: { task: Task } ) => {
 		<li
 			className={ classnames( 'launchpad__task', {
 				'keep-active': keepActive,
-				'is-completed': isCompleted,
+				'is-completed': isCompleted && ! keepActive,
+				pending: ! isCompleted,
```

Comment:
I created a `pending` classname for the tasks that haven't been completed yet. This way we will have a corresponding classname for each of the three statuses: completed, pending and completed tasks that are kept active (tasks you can revisit). This allows for much easier styling since now each task type will have only one classname as opposed to before where we had a task both `is-complete` and `keep-active` which would cause style collisions

Reply:
> This allows for much easier styling since now each task type will have only one classname as opposed to before where we had a task both is-complete and keep-active which would cause style collisions

What happens when `keepActive` is `true` and `isCompleted` is `false`? It looks like we'll have both the `keep-active` and `pending` classnames applied at the same time, which seems to be at odds with the purpose of these changes?

Reply:
@jeyip Great catch about the extra check, I tried to shorten the expression but it's definitely needed. I like the idea of inline documentation as well, I'm starting on these

Reply:
I will test just to confirm this works shortly, but also wanted to add a general note. 

I've mentioned in a few other places that it seems like we have a lot of different, related states for completion/active/disabled/etc all interacting, and that the number of states and the different interactions can be quite hard to follow.

Much of that existed before this PR. But the code in question here seems to illustrate it, and I'm wondering if there is a way to simplify this?

Example questions:

```
'keep-active': keepActive && isCompleted, // a task that is completed and can be revisited
'is-completed': isCompleted && ! keepActive, // a task that is completed and can't be revisited
pending: ! isCompleted && ! keepActive, // a task that hasn't been completed yet
```

- We have a keepActive state and keep-active css class. But these don't mean the same thing, because the keep-active css class also requires that isCompleted is true. I would think keep-active would mean keep active regardless of completion state?
- We have an isCompleted state and and is-completed css class, but these don't mean the same thing, because the isCompleted css class also requires keepActive to be false. I would think isCompleted would mean a task is completed, regardless of keepActive state. 
- We're now adding pending as a css class, which is a bit like adding yet another potential state. It says 'a task that has not been completed yet', but would have thought that was basically the default for any task? 
- We also still have taskDisabled, which is not used here in the css classes, but seems to overlap with these states in unclear ways. 

I'm not entirely sure what the solution is here. I think at root, we really still need to simplify the various states for each task. But one though for now for now would this: 
 - apply keep-active when keepActive === true
 - apply is-complete when isComplete === true
 - handle styling for combinations via css rules (ie, .keep-active.is-complete to handle situations when both are true)

Reply:
Thanks for taking a look at this @edanzer. I agree that at this point the statuses are created with styling in mind but I think that's fine because that's their only purpose. The reason I decided to add the `pending` classname was because there wasn't an easy way to add a style that is exclusive to only the tasks that are still not completed. We had to target the `.launchpad__task` but that would spill over both the `.is-completed` and `.keep-active` tasks. Then we would have to override the unwanted styles in both of these classes. It's the same when you want certain styles for the `.is-completed` tasks but not the `.keep-active` tasks. Since by default a `.keep-active` task is a `.is-completed` task as well. The benefit of going with the proposed approach is that we can have clean exclusive styles for each task status that won't spill over to the other tasks by default. I believe in the end it's a tradeoff between better semantics and hard styling or worse semantics and better styling. Curious to see your thoughts on this as well @Addison-Stavlo @jeyip

Reply:
> Since by default a .keep-active task is a .is-completed task as well. The benefit of going with the proposed approach is that we can have clean exclusive styles for each task status that won't spill over to the other tasks by default. I believe in the end it's a tradeoff between better semantics and hard styling or worse semantics and better styling. Curious to see your thoughts on this as well @Addison-Stavlo @jeyip

Part of the confusion here is the naming we're using.

```
'keep-active': keepActive && isCompleted
'is-completed': isCompleted && ! keepActive
```

We could probably rename things to help with the confusion. An idea for the former might be `completed-and-active`, while the latter might be `completed-and-inactive`, or they could be something else entirely.

Reply:
> Can we create a follow-up task to step back and re-evaluate how to simplify the different task states?

@Sharpirate Were you make a follow-up task for this?